src/plugins/dashboard/public/services/dashboard_backup/dashboard_backup_service.ts

```diff
@@ -125,7 +131,10 @@ class DashboardBackupService implements DashboardBackupServiceType {
         )
           dashboardsWithUnsavedChanges.push(dashboardId);
       });
-      return dashboardsWithUnsavedChanges;
+
+      return isEqual(oldDashboardsWithUnsavedChanges, dashboardsWithUnsavedChanges)
+        ? oldDashboardsWithUnsavedChanges
+        : dashboardsWithUnsavedChanges;
```

Comment:
This prevents an infinite React loop that would happen when an error was caught in the `DashboardUnsavedListing` component.